### DISCUSSION TITLE
GC9A01 initialisation displays random data on screen until blanked by application

### DISCUSSION THREAD

**Discussion by jfinnie** [Bodmer/TFT_eSPI#2559] [General]
GC9A01 controller doesn't clear its buffer memory on startup, and it is speficied as having random contents at power up, and not being cleared by reset.

This means that at initialisation will see either static briefly, or in the case of a warm boot, the previously displayed image content.
To avoid this, you could clear the full image memory before enabling the display output (command 0x29)

I did it as follows in another implementation, not sure how it would best fit within your architecture.  But the key really is not sending 0x29 within the initialisation code as it's inevitable that if you do this you will display random or stale contents until the application is able to re-write the display.

```
// make sure controller blanks screen, probably unnecessary
    GC9A01_write_command(0x28);
    // blank memory
    GC9A01_write_command(0x2C);
    GC9A01_set_data_command(1);
    GC9A01_set_chip_select(0);
    for (int x = 0; x < 240*240*3; x++) {
        GC9A01_spi_tx(0x00, 1);
    }
    GC9A01_set_chip_select(1);
    // unblank screen
    GC9A01_write_command(0x29);
```

**Comment by Bodmer** [comment]
Thanks for the tip. The other option is to keep the backlight LED off until init is complete.

**Reply by jfinnie** [reply]
Thanks, yes, you're right, and it works a lot better on these TFTs than on displays I've been involved with in the past.  It's probably still "the right thing to do" to ensure the memory is clear before enabling the display output (not everyone will have backlight under SW control), but it does pretty much completely mitigate the effect on the one module I have access to.
